Analysis/Tasks/PWGCF/FemtoDream/include/FemtoDream/FemtoDreamContainer.h

```diff
@@ -60,80 +64,43 @@ class FemtoDreamContainer
   {
     float femtoObs;
     if (mFemtoObs == femtoDreamContainer::Observable::kstar) {
-      femtoObs = getkstar(part1, mMassOne, part2, mMassTwo);
+      femtoObs = mFemtoDreamMath->getkstar(part1, mMassOne, part2, mMassTwo);
     }
-    const float kT = getkT(part1, mMassOne, part2, mMassTwo);
+    const float kT = FemtoDreamMath::getkT(part1, mMassOne, part2, mMassTwo);
+    const float mT = FemtoDreamMath::getmT(part1, mMassOne, part2, mMassTwo);
 
     if (mHistogramRegistry) {
       mHistogramRegistry->fill(HIST("relPairDist"), femtoObs);
       mHistogramRegistry->fill(HIST("relPairkT"), kT);
+      mHistogramRegistry->fill(HIST("relPairkstarkT"), femtoObs, kT);
+      mHistogramRegistry->fill(HIST("relPairkstarmT"), femtoObs, mT);
     }
   }
 
   template <typename T>
-  static float getkstar(const T& part1, const float mass1, const T& part2, const float mass2)
+  void SetEventPairHistograms(float multEvt, T const& part1, T const& part2)
```

Comment:
this I would do completely differently.
I would have one function called for each event
SetEventProperties(float mult, ...) that sets some member variables of this class (similarly to the mass)
and then in the setPair function you use these variables

Reply:
otherwise you need to compute kstar twice per pair etc.

Reply:
Ok. Now I also understand the comment above.